src/rgw/driver/sfs/sqlite/sqlite_buckets.cc

```diff
@@ -97,4 +97,42 @@ std::vector<std::string> SQLiteBuckets::get_deleted_buckets_ids() const {
   );
 }
 
+const std::optional<SQLiteBucketStats> SQLiteBuckets::get_stats(
+    const std::string& bucket_id
+) const {
+  auto storage = conn->get_storage();
+  std::optional<SQLiteBucketStats> stats;
+  storage.transaction([&]() mutable {
+    // ensure bucket exists
+    auto tmp = storage.get_all<DBBucket>(
+        where(is_equal(&DBBucket::bucket_id, bucket_id))
+    );
+    if (tmp.size() == 0) {
+      return false;
+    }
+    ceph_assert(tmp.size() == 1);
+
+    stats = SQLiteBucketStats();
+    stats->obj_count = storage.count<DBOPObjectInfo>(
+        where(is_equal(&DBOPObjectInfo::bucket_id, bucket_id))
+    );
+
+    auto res = storage.sum(
+        &DBVersionedObject::size,
+        left_join<DBOPObjectInfo>(
+            on(is_equal(&DBOPObjectInfo::uuid, &DBVersionedObject::object_id))
+        ),
+        where(is_equal(&DBOPObjectInfo::bucket_id, bucket_id))
+    );
+
+    if (res) {
+      stats->size = *res;
+    }
+
+    return true;
+  });
```

Comment:
This could probably be one sql statement with storage.select()?
Does RGW call this a lot?

pseudo sql:
```sql
select 
count(object_id) as obj_count,
sum(size) as size,
from buckets, objects, versioned objects
where buckets.bucket_id = objects.bucket.id and objects.uuid = versioned_object.object_id
group by object_id
```